lib/consul/async/consul_endpoint.rb

```diff
@@ -233,6 +242,13 @@ def fetch
           connect_timeout: 5, # default connection setup timeout
           inactivity_timeout: conf.wait_duration + 1 + (conf.wait_duration / 16) # default connection inactivity (post-setup) timeout
         }
+        unless conf.tls_client_cert.nil?
+          options[:tls] = {
+            cert_chain_file: conf.tls_client_cert,
+            private_key_file: conf.tls_client_key,
+            verify_peer: false # TODO: use the tls_cacert in the cert_chain_file and verify
```

Comment:
Not OK, I won't merge this.
If you want to have this behavior, add a flag (defaulting to false) called `--skip-tls-verify-peer`and use this flag (with help describing it is dangerous) to feed this field. But I don't want it to be default or too easy to activate by default